go/worker/keymanager/churp.go

```diff
@@ -0,0 +1,471 @@
+		initCh:      make(chan struct{}),
+		kmWorker:    kmWorker,
+		churps:      churps,
+		submissions: newSubmissionScheduler(kmWorker),
+	}, nil
+}
+
+// Methods implements RPCAccessController interface.
+func (w *churpWorker) Methods() []string {
+	var methods []string
+	methods = append(methods, maps.Keys(secureChurpRPCMethods)...)
+	methods = append(methods, maps.Keys(insecureChurpRPCMethods)...)
+	return methods
+}
+
+// Connect implements RPCAccessController interface.
+func (w *churpWorker) Connect(core.PeerID) bool {
+	return false
+}
+
+// Authorize implements RPCAccessController interface.
+func (w *churpWorker) Authorize(string, enclaverpc.Kind, core.PeerID) error {
+	return nil
+}
+
+// Initialized returns a channel that will be closed when the worker
+// is initialized.
+func (w *churpWorker) Initialized() <-chan struct{} {
+	return w.initCh
+}
+
+// GetStatus returns the worker status.
+func (w *churpWorker) GetStatus() workerKm.ChurpStatus {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	status := workerKm.ChurpStatus{
+		Schemes: make(map[uint8]workerKm.ChurpSchemeStatus),
+	}
+
+	for id, st := range w.churps {
+		status.Schemes[id] = workerKm.ChurpSchemeStatus{
+			Status: st,
+		}
+	}
+
+	return status
+}
+
+func (w *churpWorker) work(ctx context.Context, _ host.RichRuntime) {
+	w.logger.Info("starting worker")
+
+	stCh, stSub := w.kmWorker.backend.Churp().WatchStatuses()
+	defer stSub.Close()
+
+	epoCh, epoSub, err := w.kmWorker.commonWorker.Consensus.Beacon().WatchEpochs(ctx)
+	if err != nil {
+		w.logger.Error("failed to watch epochs",
+			"err", err,
+		)
+		return
+	}
+	defer epoSub.Close()
+
+	blkCh, blkSub, err := w.kmWorker.commonWorker.Consensus.WatchBlocks(ctx)
+	if err != nil {
+		w.logger.Error("failed to watch blocks",
+			"err", err,
+		)
+		return
+	}
+	defer blkSub.Close()
+
+	close(w.initCh)
+
+	for {
+		select {
+		case epoch := <-epoCh:
+			w.handleNewEpoch(epoch)
+		case blk := <-blkCh:
+			w.handleNewBlock(ctx, blk)
+		case status := <-stCh:
+			w.handleStatusUpdate(status)
+		case <-ctx.Done():
+			w.logger.Info("stopping worker")
+			w.submissions.Stop()
+			return
+		}
+	}
+}
+
+// handleNewEpoch is responsible for handling a new epoch.
+func (w *churpWorker) handleNewEpoch(epoch beacon.EpochTime) {
+	w.submissions.Cancel(epoch)
+	w.submissions.Clear(epoch)
+}
+
+// handleNewBlock is responsible for handling a new block.
+func (w *churpWorker) handleNewBlock(ctx context.Context, blk *consensus.Block) {
+	w.submissions.Start(ctx, blk.Height)
+}
+
+// handleStatusUpdate is responsible for handling status update.
+func (w *churpWorker) handleStatusUpdate(status *churp.Status) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	// Skip schemes we are not involved in.
+	if status.RuntimeID != w.kmWorker.runtimeID {
+		return
+	}
+	if _, ok := w.churps[status.ID]; !ok {
+		return
+	}
+	w.churps[status.ID] = status
+
+	w.logger.Debug("handle status update",
+		"status", fmt.Sprintf("%+v", status),
```

Comment:
Does simply passing it as `status` not JSON-encode the output when using JSON logs?

Reply:
It does.